2019-03-Autumn/1_linked_lists/README.md

```diff
@@ -36,41 +36,74 @@ If the problem is simple enough, remove this section. TODO :bug:
 
 TODO :bug:
 
-### 3. PROBLEM 3 TODO :bug:
+### 3. Sum Lists
 
-Source: TODO :bug:
+Source: CTCI 2.5
 
 #### Scenario
 
-Problem Statement TODO :bug:
+You have two integers, each represented by a linked list where each node 
+contains a single digit. The digits are stored in _reverse_ order, such 
+that the 1's digit is at the head of the list. Write a function that adds 
+the two numbers and returns the sum as a linked list.
 
-#### Example Input
+**Example:**
 
-If the problem is simple enough, remove this section. TODO :bug:
+Input: `( 7 -> 1 -> 6 )  +  ( 5 -> 9 -> 2 )`. That is, `617 + 295`.
 
-#### Function Signature
+Output: `( 2 -> 1 -> 9 )`. That is, `912`.
 
-TODO :bug:
+###### Bonus Question (If You Have Time)
 
-## Solutions
+Suppose the digits are stored in forward order. Repeat the above problem. 
 
-### 1. PROBLEM 1 TODO :bug:
+**Example:**
 
-Source: TODO :bug:
+Input: `( 6 -> 1 -> 7 )  +  ( 2 -> 9 -> 5 )`. That is, `617 + 295`.
 
-#### Naive/Simple Solution
+Output: `( 9 -> 1 -> 2 )`. That is, `912`.
 
-TODO :bug:
+#### Function Signature
 
-#### Optimal Solution
+Java:
 
-TODO :bug:
+```java
+public class Node {
+    public int data;
+    public Node next;
+}
 
-#### Testing The Solutions OR Driver For Solution
+public Node sumReversedNums(Node n1, Node n2) {
+    // your code here
+}
 
-TODO :bug:
+// bonus
+public Node sumOrderedNums(Node n1, Node n2) {
+    // your code here
+}
+```
 
-### 2. PROBLEM 2 TODO :bug:
+C++:
+
+```c++
+struct Node {
+    int data;
+    Node* next;
+}
+
```

Comment:
};

Reply:
Will address this.

Pro tip: if you want to suggest a code change in-line that can be committed by the PR author from within GitHub's UI, you can click the little +/- button (see right button):

![suggest_changes](https://user-images.githubusercontent.com/29528527/66527312-894a4380-eab0-11e9-90a2-bb796993617a.png)

It will throw you into an editor that looks like:
````
```suggestion
This is the old line you wanted to comment on
```
````

From there, you can edit code to your heart's desire. (Note that you yourself cannot commit the change to the PR's branch; only the PR author can do that.)